src/CitizenProject/CitizenProjectListener.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace AppBundle\CitizenProject;
+
+use AppBundle\Entity\CitizenProject;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use League\Flysystem\Filesystem;
+use League\Glide\Server;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class CitizenProjectListener
+{
+    /**
+     * @var Filesystem
+     */
+    private $storage;
+
+    /**
+     * @var Server
+     */
+    private $glide;
+
+    public function __construct(Filesystem $storage, Server $glide)
+    {
+        $this->storage = $storage;
+        $this->glide = $glide;
+    }
+
+    public function preUpdate(CitizenProject $citizenProject, LifecycleEventArgs $event)
+    {
+        $this->addImage($citizenProject);
+    }
+
+    public function prePersist(CitizenProject $citizenProject, LifecycleEventArgs $event)
+    {
+        $this->addImage($citizenProject);
+    }
+
+    private function addImage(CitizenProject $citizenProject): void
+    {
+        // Save citizen project image to cloud storage
+        if ($citizenProject->getImage() instanceof UploadedFile) {
+            $pathImage = $citizenProject->getImagePath();
+            $this->storage->put($pathImage, file_get_contents($citizenProject->getImage()->getPathname()));
+            $this->glide->deleteCache($pathImage);
+            $path = $this->glide->makeImage($pathImage, [
+                'fm' => 'jpeg',
+            ]);
+            $this->storage->put($pathImage, $this->glide->getCache()->readStream($path));
+            $this->glide->deleteCache($pathImage);
+            $citizenProject->setImageUploaded(true);
```

Comment:
I noticed this boolean is used on other entities but I don't understand what is the usefulness of this ?